packages/js/src/ui/components/Notification/DefaultNotification.tsx

```diff
@@ -14,20 +16,49 @@ const NotificationBody = (props: NotificationBodyProps) => {
 };
 
 type DefaultNotificationProps = {
-  notification: InboxNotification;
+  notification: Notification;
+  onNotificationClick?: NotificationClickHandler;
+  onActionClick?: NotificationActionClickHandler;
 };
 
 //TODO: Complete the implementation
 export const DefaultNotification = (props: DefaultNotificationProps) => {
   const style = useStyle();
   const { t, locale } = useLocalization();
 
+  const handleNotificationClick: JSX.EventHandlerUnion<HTMLAnchorElement, MouseEvent> = (e) => {
+    e.stopPropagation();
+    e.preventDefault();
+
+    if (!props.notification.isRead) {
+      props.notification.read();
+    }
+
+    props.onNotificationClick?.({ notification: props.notification });
+    if (props.notification.redirect?.url) {
+      window.open(props.notification.redirect?.url, '_blank', 'noreferrer noopener');
+    }
+  };
```

Comment:
when the notification is clicked:
- mark it as read if is not already
- call the `onNotificationClick` prop
- if the `redirect.url` was passed, open the link in a new browser tab

Reply:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
💭 **thought (non-blocking):** On a second thought, do we really need the redirectUrl feature?
We have the `onNotificationClick`, users can add their own logic to `navigate/redirect` using their own implementation of router. and also will be easier for them to create on demand Urls in their app.‏

Reply:
@BiswaViraj https://github.com/novuhq/novu/pull/6227/files#r1702356457
We provide flexibility in building the dynamic URL based on the trigger payload, system variables etc., and now with the framework they can do that by querying their db or any other system. This way on the FE they don't need to fetch the data from API to build. But also, it might not be possible to recreate the same context/state of the system similarly to which they did the trigger (I mean when fetching the data from API). 

I've checked the DB and half of all redirect URLs use the variables.